crates/subspace-farmer/src/dsn.rs

```diff
@@ -141,11 +101,15 @@ where
                 }
             })
             .await
-            .expect("`on_pieces` must never panic")
-            {
-                return result;
-            }
+            .expect("`on_pieces` must never panic")?;
         }
+
+        // In case we ended at `MAX` we want to start from 0 next time
+        offset = offset.saturating_add(if end == PieceIndexHashNumber::MAX {
+            end - start + 1
```

Comment:
Reaching this line means we'll sync across the multiple plots, right?

Reply:
No, it means that we want to sync a plot like this:

```
0                        MAX
|-------------------------|
|-----|              |***-|
|-------------------------|
```
Stars there mean the part which is already synced, while dashes are the part which is not yet.

This case is kinda special, cause we need to finish up the last sector which is `x..MAX` and make next sector to start from 0.

Reply:
I see, thank you!